tests/s3_bucket/test_github_content_compare_slugs_styles.py

```diff
@@ -0,0 +1,107 @@
+import requests
+
+import urllib.parse
+from urllib.error import HTTPError
+
+import pytest
+
+from bs4 import BeautifulSoup
+
+
+"""
+Compares slugs/styles between books.xml and ABL file of every collection in github content repo.
+Latest update on June 6th, 2022
+"""
+
+
+def test_github_content_compare_slugs_styles(
+    git_content_repos, headers_data, abl_books_slugs_styles
+):
+
+    for repo in git_content_repos:
+
+        print("\nNow verifying: ", repo)
+
+        meta_inf_dir = f"https://api.github.com/repos/openstax/{repo}/contents/META-INF/"
+
+        meta_inf_list = requests.get(meta_inf_dir, headers=headers_data)
+
+        if meta_inf_list.status_code != 200:
+
+            # Return code 404, 501, ... for incorrect meta-inf url
+            print(f">>>>> FAILED {meta_inf_list.status_code}: no meta-inf folder in {repo}")
+
+        else:
+
+            for item in meta_inf_list.json():
+
+                if item["type"] != "file":
+
+                    # Ignore anything that may not be a file
+                    continue
+
+                else:
+
+                    rel_path = urllib.parse.quote(item["path"])
+                    meta_inf_url = (
+                        f"https://api.github.com/repos/openstax/{repo}/contents/{rel_path}"
+                    )
+
+                    try:
+
+                        meta_inf_resp = requests.get(meta_inf_url, headers=headers_data)
+
+                    except HTTPError as h_e:
+                        # Return code 404, 501, ... for incorrect/missing books.xml
+                        pytest.fail(
+                            f"HTTP Error {h_e.code}: incorrect/missing books.xml {meta_inf_url}"
+                        )
+
+                    else:
+
+                        resp_content = meta_inf_resp.text
```

Comment:
This request [returns a json document](https://docs.github.com/en/rest/repos/contents). Within that json document is a property called `contents` that is base64 encoded. You can decode the value of this property and use the xml document that way:

```python
from base64 import b64decode
json_resp = meta_inf_resp.json()
resp_content = b64decode(json_resp["content"])
```

Reply:
@TylerZeroMaster I tried `json_resp = meta_inf_resp.json()` before but it gives an error (seems like not a json), so I went with .text which works...

Reply:
I don't know what to say. I ran this locally step-by-step and the result was...
```python
>>> meta_inf_resp.text
'{"name":"books.xml","path":"META-INF/books.xml","sha":"71b6b7d577e5b4180e4e8750da622c9f52bcbe88","size":368,"url":"https://api.github.com/repos/openstax/osbooks-chemistry-bundle/contents/META-INF/books.xml?ref=main","html_url":"https://github.com/openstax/osbooks-chemistry-bundle/blob/main/META-INF/books.xml","git_url":"https://api.github.com/repos/openstax/osbooks-chemistry-bundle/git/blobs/71b6b7d577e5b4180e4e8750da622c9f52bcbe88","download_url":"https://raw.githubusercontent.com/openstax/osbooks-chemistry-bundle/main/META-INF/books.xml","type":"file","content":"PGNvbnRhaW5lciB4bWxucz0iaHR0cHM6Ly9vcGVuc3RheC5vcmcvbmFtZXNw\\nYWNlcy9ib29rLWNvbnRhaW5lciIgdmVyc2lvbj0iMSI+CiAgICA8Ym9vayBz\\nbHVnPSJjaGVtaXN0cnktMmUiIGNvbGxlY3Rpb24taWQ9ImNvbDI2MDY5IiBz\\ndHlsZT0iY2hlbWlzdHJ5IiBocmVmPSIuLi9jb2xsZWN0aW9ucy9jaGVtaXN0\\ncnktMmUuY29sbGVjdGlvbi54bWwiIC8+CiAgICA8Ym9vayBzbHVnPSJjaGVt\\naXN0cnktYXRvbXMtZmlyc3QtMmUiIGNvbGxlY3Rpb24taWQ9ImNvbDI2NDg4\\nIiBzdHlsZT0iY2hlbWlzdHJ5IiBocmVmPSIuLi9jb2xsZWN0aW9ucy9jaGVt\\naXN0cnktYXRvbXMtZmlyc3QtMmUuY29sbGVjdGlvbi54bWwiIC8+CjwvY29u\\ndGFpbmVyPgo=\\n","encoding":"base64","_links":{"self":"https://api.github.com/repos/openstax/osbooks-chemistry-bundle/contents/META-INF/books.xml?ref=main","git":"https://api.github.com/repos/openstax/osbooks-chemistry-bundle/git/blobs/71b6b7d577e5b4180e4e8750da622c9f52bcbe88","html":"https://github.com/openstax/osbooks-chemistry-bundle/blob/main/META-INF/books.xml"}}'
```

Reply:
It was because of the header `Accept: application/vnd.github.v3.raw` in headers_data fixture. This works as you had it.